### PR TITLE
Faster join on high join timeouts

### PR DESCRIPTION
**Pull request by surfdude75**
There is no need to wait timeout to check if joined.

### CODE REVIEW

File: lib/swim.js [hunk 1]
```diff
@@ -140,34 +140,52 @@ Swim.prototype.join = function join(hosts, callback) {
...
-        if (numberOfHostsResponded >= 1) {
-            if (typeof callback === 'function') {
-                callback();
-            } else {
-                self.emit(Swim.EventType.Ready);
-            }
-        } else {
-            err = new JoinFailedError({
+    var timeout = setTimeout(function joinTimeout() {
+        if (checker) {
```
Comment (mrhooray) [inline lib/swim.js]: if i'm not mistaken `clearInternal` can take `null`

File: lib/swim.js [hunk 2]
```diff
@@ -140,34 +140,52 @@ Swim.prototype.join = function join(hosts, callback) {
...
-
             if (typeof callback === 'function') {
                 callback(err);
             } else {
                 self.emit(Swim.EventType.Error, err);
             }
         }
     }, self.joinTimeout);
+
+    var checkerInterval = 50;
```
Comment (mrhooray) [inline lib/swim.js]: shall we move and expose this as configuration

File: lib/swim.js [hunk 3]
```diff
@@ -140,34 +140,52 @@ Swim.prototype.join = function join(hosts, callback) {
...
+    }
+
+    function checkJoin() {
+        var numberOfHostsResponded = hosts.reduce(function countRespondedHosts(num, host) {
+            num += self.membership.get(host) ? 1 : 0;
+            return num;
+        }, 0);
+        if (numberOfHostsResponded >= 1) {
+            if (checker) {
+                clearInterval(checker);
```
Comment (mrhooray) [inline lib/swim.js]: ditto

**Comment by mrhooray** [comment]
Thanks @surfdude75 and sorry for the delay.
We did not add this as low join timeout is usually used. It's indeed an issue for higher values.

A few minor comments on the change.

**Comment by surfdude75** [comment]
Thanks for your attention @mrhooray and for the helpful open source project.

More the adjusments are ok.

Let me know if you need something else.

Aloha.

**Comment by mrhooray** [comment]
will tag a new release for this

**Pull request merged**